transform.go

```diff
@@ -98,8 +104,21 @@ func (t *Transformer) Transform(doc *ast.Document, reader text.Reader, ctx parse
 		title = _defaultTitle
 	}
 
+	titleDepth := t.TitleDepth
+	if titleDepth < 1 {
+		titleDepth = _defaultTitleDepth
+	}
+
+	// Defaults to max title depth of 6
+	// if given TitleDepth > 6. Could also
+	// make sense to default to
+	// _defaultTitleDepth instead.
```

Comment:
Na, what you have is right. If they say 7, we max it out at 6.
Either that, or error, and I don't think we should error in this API.